src/new/sendTransaction.js

```diff
@@ -193,6 +193,11 @@ class SendTransaction extends EventEmitter {
     }
 
     const walletData = wallet.getWalletData();
+    if (walletData === null) {
```

Comment:
It is a race condition, right? Nice catch! What if a new wallet is started in the between? Would it add this transaction to the history of the wrong wallet?

Reply:
Yeah, this is a condition I caught in the app when I send the transaction, it succeeds, then I reset the wallet before the `setTimeout` is triggered. Then the `walletData` would be `null`.

We have this `if (input.tx_id in historyTransactions)` and if it's the wrong wallet we wouldn't set this value.

Reply:
Don't we block the UX until the transaction is broadcasted to the network?

Reply:
Yeah, but the `setTimeout` is execute 60s after the tx is sent to be mined. We could clear the `setTimeout` on success as well. But I would keep this protection anyway to prevent any race condition.

Reply:
Ok. I agree to keep this protection here but we definitely should clear the `setTimeout` on success.

Reply:
Done